platform/core/src/services/WorkflowStagesService/WorkflowStagesService.ts

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react';
+import { CommandsManager } from '../../classes';
+import { ServicesManager, ToolbarService } from '../../services';
+import { PubSubService } from '../_shared/pubSubServiceInterface';
+
+export const EVENTS = {
+  ACTIVE_STAGE_CHANGED: 'event::workflowStagesService:activateStageChanged',
+  STAGES_CHANGED: 'event::workflowStagesService:stagesChanged',
+};
+
+export type WorkflowStage = {
+  id: string;
+  name: string;
+  toolbar?: {
+    buttons: unknown[];
+    sections: {
+      key: string;
+      buttons: string[];
+    }[];
+  };
+  hangingProtocol?: {
+    protocolId: string;
+    stageId?: string;
+  };
+  layout?: {
+    panels: {
+      leftPanels?: unknown[];
+      rightPanels?: unknown[];
+      rightPanelDefaultClosed?: boolean;
+    };
+  };
+};
+
+class WorkflowStagesService extends PubSubService {
+  private _servicesManager: ServicesManager;
+  private _commandsManager: CommandsManager;
+  private _stages: WorkflowStage[];
+  private _activeStage: WorkflowStage;
+
+  constructor(
+    commandsManager: CommandsManager,
+    servicesManager: ServicesManager
+  ) {
+    super(EVENTS);
+    this._stages = [];
+    this._activeStage = null;
```

Comment:
I would suggest no to use stage without workflowStage, since we have hangingprotocol stage too